app/src/main/java/org/mozilla/focus/cfr/CfrMiddleware.kt

```diff
@@ -36,20 +39,30 @@ class CfrMiddleware(private val components: Components) : Middleware<BrowserStat
             }
         }
 
-        if (shouldShowCfrForTrackingProtection(action = action)) {
+        if (shouldShowCfrForTrackingProtection(action = action, browserState = context.state)) {
             components.appStore.dispatch(AppAction.ShowTrackingProtectionCfrChange(true))
         }
     }
 
+    // We shouldn't show CFR for Mozilla pages #6383.
```

Comment:
nit: This comment doesn't really apply to the `isMozillaUrl` method which doesn't control whether the CFR would be shown or not. It's purpose is just to check the etld domain.

Reply:
Done